evil-commands.el

```diff
@@ -3742,6 +3742,18 @@ The 'bang' argument means to sort in reverse order."
 
 ;;; Window navigation
 
+(defmacro evil-save-side-windows (&rest body)
+  "Toggle side windows, evaluate BODY, restore side windows."
+  (declare (indent defun) (debug (&rest form)))
+  (let ((sides (make-symbol "sidesvar")))
+    `(let ((,sides (and (functionp 'window-toggle-side-windows)
+                        (window-with-parameter 'window-side))))
+       (when ,sides
+         (window-toggle-side-windows))
+       ,@body
+       (when ,sides
+         (window-toggle-side-windows)))))
```

Comment:
Shouldn't this macro use a `condition-case` block?

Reply:
Yup your right there does need to be some error handling.